### ISSUE TITLE
CMake support

### ISSUE THREAD

**Issue by wonder-mice** [joyent/http-parser#257]
Hi, I'm going to write CMakeLists.txt for http-parser to use it in my project.
I could put a bit more effort into it (add tests support, install step, etc.) if you are interested in merging CMake support in.
I see that http-parser also has Make and GYP support, and having third build system could be too much.
What do you think about it?

**Comment by Lupus** [comment]
+1

2015-07-02 21:45 GMT+03:00 wonder-mice <notifications@github.com>:

> Hi, I'm going to write CMakeLists.txt for http-parser to use it in my
> project.
> I could put a bit more effort into it (add tests support, install step,
> etc.) if you are interested in merging CMake support in.
> I see that http-parser also has Make and GYP support, and having third
> build system could be too much.
> What do you think about it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joyent/http-parser/issues/257>.
>



-- 
Regards,
Konstantin

**Comment by indutny** [comment]
Considering that there not going to be much build changes, I'm totally fine with it. Please go ahead! ;)

Thanks.

**Comment by nibua-r** [comment]
+1

**Comment by jasnell** [comment]
@wonder-mice ... is this still something you're interested in?

**Comment by RWOverdijk** [comment]
:+1:

**Comment by wonder-mice** [comment]
Yes, I have it almost ready, but can't find enough time to groom it before publishing.

**Comment by rfdickerson** [comment]
Similarly, any interest in a pkg-config file? (.pc)?

**Comment by piribes** [comment]
Any news ?

**Comment by flandr** [comment]
Sent out a pull request with a minimal CMake build. It only provides static and shared instances of the main library target, but I think it will be more productive to start basic and iterate from there rather than waiting for a complete port of all the Makefile targets, etc.
